source/shallow_water/initial_state_geotiff.h

```diff
@@ -0,0 +1,395 @@
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// Copyright (C) 2024 - 2024 by the ryujin authors
+//
+
+#pragma once
+
+#include <initial_state_library.h>
+#include <lazy.h>
+
+#include <deal.II/base/function_parser.h>
+
+
+#ifdef WITH_GDAL
+#include <cpl_conv.h>
+#include <gdal.h>
+#include <gdal_priv.h>
+#endif
+
+namespace ryujin
+{
+  namespace ShallowWaterInitialStates
+  {
+    /**
+     * Returns an initial state defined by a set of user specified functions
+     * based on the primitive variables.
+     *
+     * @ingroup ShallowWaterEquations
+     */
+    template <typename Description, int dim, typename Number>
+    class GeoTIFF : public InitialState<Description, dim, Number>
+    {
+    public:
+      using HyperbolicSystem = typename Description::HyperbolicSystem;
+      using View =
+          typename Description::template HyperbolicSystemView<dim, Number>;
+      using state_type = typename View::state_type;
+
+
+      GeoTIFF(const HyperbolicSystem &hyperbolic_system,
+              const std::string subsection)
+          : InitialState<Description, dim, Number>("geotiff", subsection)
+          , hyperbolic_system_(hyperbolic_system)
+      {
+        filename_ = "ryujin.tif";
+        this->add_parameter(
+            "filename", filename_, "GeoTIFF: image file to load");
+
+        transformation_ = {0., 0.01, 0., 0., 0., 0.01};
```

Comment:
Out of curiosity, where did these numbers come from?

Reply:
The numbers are simply meant to have a nonsingular transformation - no deeper meaning. I chose to scale the x and y axis by 0.01 so that the total dimension of the domain does not become too large (which might be unexpected).